zola fails to build on amd64

The package zola with version 0.20.0 and epoch 3 fails to build from source on amd64 with the following error:

```
2025-06-09 08:48:48.093777504 +0000 UTC   cargo:warning=  232 | #define st_cleanup_safe             onig_st_cleanup_safe
2025-06-09 08:48:48.093780794 +0000 UTC   cargo:warning=      |                                     ^~~~~~~~~~~~~~~~~~~~
2025-06-09 08:48:48.093783851 +0000 UTC   cargo:warning=oniguruma/src/st.c:477:1: note: in expansion of macro 'st_cleanup_safe'
2025-06-09 08:48:48.093787046 +0000 UTC   cargo:warning=  477 | st_cleanup_safe(table, never)
2025-06-09 08:48:48.093790246 +0000 UTC   cargo:warning=      | ^~~~~~~~~~~~~~~
2025-06-09 08:48:48.09379357 +0000 UTC   cargo:warning=oniguruma/src/st.c:483:21: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025-06-09 08:48:48.093799534 +0000 UTC   cargo:warning=  483 |   st_foreach(table, delete_never, never);
2025-06-09 08:48:48.09380324 +0000 UTC   cargo:warning=      |                     ^~~~~~~~~~~~
2025-06-09 08:48:48.093806362 +0000 UTC   cargo:warning=      |                     |
2025-06-09 08:48:48.093809519 +0000 UTC   cargo:warning=      |                     int (*)(st_data_t,  st_data_t,  st_data_t) {aka int (*)(long unsigned int,  long unsigned int,  long unsigned int)}
2025-06-09 08:48:48.093812872 +0000 UTC   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_data_t,  st_data_t,  st_data_t)' {aka 'int (*)(long unsigned int,  long unsigned int,  long unsigned int)'}
2025-06-09 08:48:48.093819513 +0000 UTC   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025-06-09 08:48:48.093824112 +0000 UTC   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025-06-09 08:48:48.093830032 +0000 UTC   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025-06-09 08:48:48.093833898 +0000 UTC   cargo:warning=   35 | # define _(args) args
2025-06-09 08:48:48.093836824 +0000 UTC   cargo:warning=      |                  ^~~~
2025-06-09 08:48:48.093843276 +0000 UTC   cargo:warning=oniguruma/src/st.c:465:1: note: 'delete_never' declared here
2025-06-09 08:48:48.093849416 +0000 UTC   cargo:warning=  465 | delete_never(st_data_t key __attribute__ ((unused)), st_data_t value,
2025-06-09 08:48:48.093854381 +0000 UTC   cargo:warning=      | ^~~~~~~~~~~~
2025-06-09 08:48:48.09386141 +0000 UTC   cargo:warning=oniguruma/src/st.c: In function 'onig_st_foreach':
2025-06-09 08:48:48.093865701 +0000 UTC   cargo:warning=oniguruma/src/regint.h:229:37: warning: old-style function definition [-Wold-style-definition]
2025-06-09 08:48:48.093869118 +0000 UTC   cargo:warning=  229 | #define st_foreach                  onig_st_foreach
2025-06-09 08:48:48.093872391 +0000 UTC   cargo:warning=      |                                     ^~~~~~~~~~~~~~~
2025-06-09 08:48:48.09387569 +0000 UTC   cargo:warning=oniguruma/src/st.c:488:1: note: in expansion of macro 'st_foreach'
2025-06-09 08:48:48.093879081 +0000 UTC   cargo:warning=  488 | st_foreach(table, func, arg)
2025-06-09 08:48:48.093882377 +0000 UTC   cargo:warning=      | ^~~~~~~~~~
2025-06-09 08:48:48.093885734 +0000 UTC   cargo:warning=oniguruma/src/st.c:500:23: error: too many arguments to function 'func'; expected 0, have 3
2025-06-09 08:48:48.093889099 +0000 UTC   cargo:warning=  500 |             retval = (*func)(ptr->key, ptr->record, arg);
2025-06-09 08:48:48.093892262 +0000 UTC   cargo:warning=      |                      ~^~~~~~ ~~~~~~~~
2025-06-09 08:48:48.093895349 +0000 UTC   cargo:warning=oniguruma/src/st.c: In function 'strhash':
2025-06-09 08:48:48.093899328 +0000 UTC   cargo:warning=oniguruma/src/st.c:538:1: warning: old-style function definition [-Wold-style-definition]
2025-06-09 08:48:48.093902624 +0000 UTC   cargo:warning=  538 | strhash(string)
2025-06-09 08:48:48.093905733 +0000 UTC   cargo:warning=      | ^~~~~~~
2025-06-09 08:48:48.093909094 +0000 UTC   cargo:warning=oniguruma/src/st.c: In function 'numcmp':
2025-06-09 08:48:48.093912394 +0000 UTC   cargo:warning=oniguruma/src/st.c:577:1: warning: old-style function definition [-Wold-style-definition]
2025-06-09 08:48:48.093915613 +0000 UTC   cargo:warning=  577 | numcmp(x, y)
2025-06-09 08:48:48.093918743 +0000 UTC   cargo:warning=      | ^~~~~~
2025-06-09 08:48:48.093921854 +0000 UTC   cargo:warning=oniguruma/src/st.c: In function 'numhash':
2025-06-09 08:48:48.093924922 +0000 UTC   cargo:warning=oniguruma/src/st.c:584:1: warning: old-style function definition [-Wold-style-definition]
2025-06-09 08:48:48.093932041 +0000 UTC   cargo:warning=  584 | numhash(n)
2025-06-09 08:48:48.093936136 +0000 UTC   cargo:warning=      | ^~~~~~~
2025-06-09 08:48:48.093939467 +0000 UTC 
2025-06-09 08:48:48.093942628 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093945631 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093948852 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093954686 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093958716 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093961824 +0000 UTC   exit status: 1
2025-06-09 08:48:48.093966896 +0000 UTC   cargo:warning=ToolExecError: Command LC_ALL="C" "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/home/build/target/release/build/onig_sys-71106757f3318d5f/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/home/build/target/release/build/onig_sys-71106757f3318d5f/out/a445302c6d3dcb51-st.o" "-c" "oniguruma/src/st.c" with args cc did not execute successfully (status code exit status: 1).
2025-06-09 08:48:48.093973853 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093977863 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093981003 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093983927 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093987414 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093990479 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093993699 +0000 UTC   exit status: 0
2025-06-09 08:48:48.093996837 +0000 UTC   exit status: 0
2025-06-09 08:48:48.09400023 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094006119 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094009929 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094016606 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094020658 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094023671 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094027254 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094030413 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094033694 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094037273 +0000 UTC   exit status: 0
2025-06-09 08:48:48.09404032 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094043346 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094046522 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094052367 +0000 UTC   exit status: 0
2025-06-09 08:48:48.09405637 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094059292 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094062323 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094069204 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094073066 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094075963 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094079664 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094083397 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094089179 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094093028 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094096348 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094099383 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094102773 +0000 UTC   exit status: 0
2025-06-09 08:48:48.09410575 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094109289 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094115164 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094119074 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094122434 +0000 UTC   exit status: 0
2025-06-09 08:48:48.094125495 +0000 UTC 
2025-06-09 08:48:48.094128529 +0000 UTC   --- stderr
2025-06-09 08:48:48.094132102 +0000 UTC 
2025-06-09 08:48:48.094135384 +0000 UTC 
2025-06-09 08:48:48.094139124 +0000 UTC   error occurred in cc-rs: Command LC_ALL="C" "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/home/build/target/release/build/onig_sys-71106757f3318d5f/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/home/build/target/release/build/onig_sys-71106757f3318d5f/out/a445302c6d3dcb51-st.o" "-c" "oniguruma/src/st.c" with args cc did not execute successfully (status code exit status: 1).
2025-06-09 08:48:48.094145754 +0000 UTC 
2025-06-09 08:48:48.094149788 +0000 UTC 
2025-06-09 08:48:48.094156396 +0000 UTC warning: build failed, waiting for other jobs to finish...
2025-06-09 08:48:57.775967944 +0000 UTC failed to build package: unable to run package zola pipeline: unable to run pipeline: unable to run pipeline: exit status 101
2025-06-09 08:48:57.861568306 +0000 UTC failed to build: command melange failed: exit status 1
2025-06-09 08:48:57.861708474 +0000 UTC build failed: command melange failed: exit status 1
```

--- a/zola.yaml
+++ b/zola.yaml
@@ -1,7 +1,7 @@
 package:
   name: zola
   version: 0.20.0
-  epoch: 3
+  epoch: 4
   description: A fast static site generator in a single binary with everything built-in
   copyright:
     - license: MIT

--- a/zola/cargobump-deps.yaml
+++ b/zola/cargobump-deps.yaml
@@ -7,3 +7,7 @@ packages:
       version: 0.17.13
     - name: tokio
       version: 1.43.1
+    # fix build error with gcc 15. remove once merged to main
+    # ref: https://github.com/getzola/zola/pull/2897/files
+    - name: onig
+      version: 6.5.1

